### PR TITLE
Publish version 1.4 

### PR DESCRIPTION
**Pull request by Ubuntufanboy**
The major changes are:

- insane syncing (MUCH better than before)
- a remote to control the video and audio of the program
- 10% faster load times

### CODE REVIEW

**Review by Ubuntufanboy — COMMENTED** [review]
Add if opener to the player when we need to make our frames

File: main.py [hunk 1]
```diff
@@ -107,20 +187,82 @@ def wk4():
...
 else:
+    # Get in directory
     os.chdir("processed")
     place = f"{x}x{y}"
     os.chdir(place)
-    Silver.play(f"{current}/badapple.mp3")
-    for i in range(1,3286):
+
+    frames = []
+    for i in range(1, 3284):
```
Comment (Debianfanboy) [inline main.py:228]: Should be for i in range(1,3286)

File: main.py [hunk 2]
```diff
@@ -65,15 +131,15 @@ def wk4():
...
 skip = False
 exists = False
 
 if p.exists():
     exists = True
-    
+
     length = sum(1 for x in p.glob('*') if x.is_file())
     if length >= 3284:
         print("Already exists")
```
Comment (Debianfanboy) [inline main.py:145]: What exists? Should mention how this is for pre-processed files

File: main.py [hunk 3]
```diff
@@ -83,19 +149,33 @@ def wk4():
         print("Too few files in directory")
 else:
     print("Directory does not exist")
```
Comment (Debianfanboy) [inline main.py:151]: What directory does not exist?

File: main.py [hunk 4]
```diff
@@ -107,20 +187,82 @@ def wk4():
...
-    for frame in frames:
-        os.system("clear")
-        print(frame)
-        time.sleep(0.0635)
+
+    next_call = time.perf_counter()
+
+    os.chdir("..")
+    os.chdir("..")
+
```
Comment (Debianfanboy) [inline main.py:195]: add ``if opener:`` here

File: main.py [hunk 5]
```diff
@@ -1,59 +1,125 @@
 import numpy as np
```
Comment (Debianfanboy) [inline main.py:1]: just in general we need more helpful comments. Add 5 more block comments and 10 more inline comments. Make sure they are actually useful

File: main.py [hunk 6]
```diff
@@ -1,59 +1,125 @@
...
             img = np.array(img)
             w4.append(process(img))
+
+
 f = Thread(target=wk1)
 u = Thread(target=wk2)
 c = Thread(target=wk3)
 k = Thread(target=wk4)
 
+print("Are you using GNOME? y/n If you don't know enter n")
```
Comment (Debianfanboy) [inline main.py:116]: Change to "If you don't know click y"

File: main.py [hunk 7]
```diff
@@ -1,59 +1,125 @@
...
+answer = input("(y/n)>>> ")
+if answer == "y":
```
Comment (Debianfanboy) [inline main.py:118]: change to is ``if answer != "n"``

File: main.py (hunk 2 above)
Comment (Debianfanboy) [reply main.py:145]: Perhaps also say "x folder has all y amount of files"

File: main.py [hunk 8]
```diff
@@ -107,20 +187,82 @@ def wk4():
...
+    # Get in directory
     os.chdir("processed")
     place = f"{x}x{y}"
     os.chdir(place)
-    Silver.play(f"{current}/badapple.mp3")
-    for i in range(1,3286):
+
+    frames = []
+    for i in range(1, 3284):
         filename = str(i) + ".txt"
```
Comment (Debianfanboy) [inline main.py:229]: This is pointless... Just change to ``with open(f"{i}.txt"):``

File: main.py [hunk 9]
```diff
@@ -107,20 +187,82 @@ def wk4():
...
         with open(filename) as f:
             frame = f.read()
```
Comment (Debianfanboy) [inline main.py:231]: This is pointless just do ``frames.append(f.read())``

**Review by Debianfanboy — CHANGES_REQUESTED** [review]
Overall, a lot of junk code. Try to remove variables. A lot of pointless code with little readability. It's important to add more comments so more people can work on the codebase

**Comment by Ubuntufanboy** [comment]
FIXED. Committing to main directly

**Pull request closed without merging**